### PR TITLE
Enhance `PodSecurity` documentation

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
Different Kubernetes versions from `v1.22` to `v1.25` use different admission config versions for `PodSecurity` admission plugin. This PR enhances the documentation with the above-mentioned.

This PR also includes one minor change in the e2e tests.

**Which issue(s) this PR fixes**:
Part of #5250 

**Special notes for your reviewer**:
/cc @rfranzke 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/usage/pod-security.md [hunk 1]
```diff
@@ -50,15 +50,17 @@ admissionPlugins:
       namespaces: []
 ```
 
-If your cluster version is `v1.22`, use `apiVersion: pod-security.admission.config.k8s.io/v1alpha1`.
-Please note that in `v1.22` the feature gate `PodSecurity` is not enabled by default. You have to add:
+⚠️ Note that `pod-security.admission.config.k8s.io/v1` configuration requires v1.25+. For v1.23 and v1.24, use `pod-security.admission.config.k8s.io/v1beta1`. For v1.22, use `pod-security.admission.config.k8s.io/v1alpha1`.
```
Comment (rfranzke) [inline docs/usage/pod-security.md]:
```suggestion
⚠️ Note that `pod-security.admission.config.k8s.io/v1` configuration requires `v1.25`+. For `v1.23` and `v1.24`, use `pod-security.admission.config.k8s.io/v1beta1`. For `v1.22`, use `pod-security.admission.config.k8s.io/v1alpha1`.
```

Nit for consistency

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Comment by shafeeqes** [comment]
I will just cherry-pick to this release because we anyway have a patch going in.
/cherrypick release-v1.60

**Pull request merged**